### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by BC0821**

### CODE REVIEW

File: src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/Powerhub.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+class PowerHub : Subsystem("PowerHub") {
+    private val roller = MotorController(
+            MotorControllers.INTAKE_A, INTAKE_B
+    ) {
+        inverted = true
+        brakeMode = true
+    }
+
+    private var intakeRunning = false
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/Powerhub.kt]:
```suggestion
    private var rollerRunning= false
```

File: src/main/kotlin/org/sert2521/infiniterecharge2020/Constants.kt [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
 import org.sert2521.sertain.motors.TalonId
+import org.sert2521.sertain.motors.VictorId
 
 object MotorControllers {
     val leftFront = TalonId(1)
     val leftBack = TalonId(3)
     val rightFront = TalonId(2)
     val rightBack = TalonId(4)
+
+    val INTAKE_A = TalonId(-1)
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/Constants.kt]:
```suggestion
    val ROLLER_LEFT = TalonId(-1)
```

File: src/main/kotlin/org/sert2521/infiniterecharge2020/Constants.kt [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+    val INTAKE_B = TalonId(-1)
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/Constants.kt]:
```suggestion
    val ROLLER_RIGHT = TalonId(-1)
```

File: src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/Powerhub.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package org.sert2521.infiniterecharge2020.powerhub
+
+import org.sert2521.infiniterecharge2020.MotorControllers
+import org.sert2521.infiniterecharge2020.MotorControllers.INTAKE_B
+import org.sert2521.sertain.motors.MotorController
+import org.sert2521.sertain.subsystems.Subsystem
+
+class PowerHub : Subsystem("PowerHub") {
+    private val roller = MotorController(
+            MotorControllers.INTAKE_A, INTAKE_B
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/Powerhub.kt]:
```suggestion
            MotorControllers.ROLLER_LEFT, ROLLER_RIGHT
```

File: src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubCommands.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            onTick {
+                periodic(20) {
+                    powerHub.spin()
+                    if (intake) {
+                        println("Trying to intake")
+                        powerHub.lowerFlapper()
+
+                    } else {
+                        println("Trying to outtake")
+                        powerHub.raiseFlapper()
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubCommands.kt]: I checked in with CAD again and apparently we don't need to set continuous power to the motor to keep the flap open, and closing the power will just require motor power in the opposite direction. Not sure if we _can_ still run the motor the whole time or if we should turn it off as soon as it opens. Can ask CAD/Mechanical more about that. If we do need to turn it off, we can probably just put it in a `timer` loop

File: src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubConstants.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package org.sert2521.infiniterecharge2020.powerhub
+
+const val INTAKE_SPEED = .9
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubConstants.kt]:
```suggestion
const val ROLLER_SPEED = .9
```

**Review by will-hou — COMMENTED** [review]
Lookin' really good! Just a few naming suggestions. We'll also need to add something in the `Oi.kt` file that will actually run the intake and outtake when a joystick button is pushed

File: src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubCommands.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    val time = System.currentTimeMillis()
+    action {
+        onTick {
+            powerhub.lowerFlapper()
+            if (System.currentTimeMillis() - time <= 1000) {
+                this@action.cancel()
+            }
+        }
+    }
+}
```
Comment (will-hou) [inline src/main/kotlin/org/sert2521/infiniterecharge2020/powerhub/PowerhubCommands.kt]: We're actually going to have a sensor telling us when the flap has reached the top/bottom positions so this command should be changed to run the flapper motor until the sensor tells us that it's reached the position.

**Review by mckelveygreg — APPROVED** [review]
Lets get this merged soon!!

**Comment by gnawinganimal** [comment]
this lgtm, therefore lgtm

**Review by gnawinganimal — APPROVED** [review]
LGTM.





Which prompts me to say, LGTM

**Pull request merged**